src/app/campaign/campaign.component.ts

```diff
@@ -1,78 +1,75 @@
+      [valid]="valid$ | async"
+      [changed]="changed$ | async"
       [isSaving]="campaignSaving$ | async"
       (save)="campaignSubmit()"
     ></grove-campaign-status>
     <mat-drawer-container autosize>
       <mat-drawer role="navigation" mode="side" opened disableClose>
-        <mat-nav-list>
-          <a mat-list-item routerLinkActive="active-link" [routerLinkActiveOptions]="{ exact: true }" routerLink="./">Campaign</a>
-          <a
-            mat-list-item
-            *ngFor="let flight of campaignFlights$ | async; let i = index"
-            [routerLink]="['flight', flight.id]"
-            routerLinkActive="active-link"
-            [class.deleted-flight]="flight.softDeleted"
-            [class.changed]="flight.changed"
-            [class.valid]="flight.valid"
-            [class.invalid]="!flight.valid"
-            [class.error]="!flight.statusOk"
-          >
-            <span matLine>{{ flight.name }}</span>
-            <mat-icon color="warn" *ngIf="!flight.statusOk">priority_high</mat-icon>
-          </a>
-        </mat-nav-list>
-        <a class="add-flight" [routerLink]="" (click)="createFlight()"><mat-icon>add</mat-icon> Add a Flight</a>
+        <grove-campaign-nav [flights]="flights$ | async" (createFlight)="createFlight()"></grove-campaign-nav>
       </mat-drawer>
       <mat-drawer-content role="main">
-        <router-outlet></router-outlet>
+        <ng-container *ngIf="campaignLoaded$ | async; else loading">
+          <div class="error" *ngIf="error$ | async as error; else content">{{ error }}</div>
+          <ng-template #content>
+            <router-outlet></router-outlet>
+          </ng-template>
+        </ng-container>
+        <ng-template #loading>
+          <div class="loading" *ngIf="campaignLoading$ | async"><mat-spinner></mat-spinner></div>
+        </ng-template>
       </mat-drawer-content>
     </mat-drawer-container>
   `,
   styleUrls: ['./campaign.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class CampaignComponent implements OnInit, OnDestroy {
-  campaignFlights$: Observable<LocalFlightState[]>;
+  flights$: Observable<FlightState[]>;
+  campaignLoaded$: Observable<boolean>;
+  campaignLoading$: Observable<boolean>;
   campaignSaving$: Observable<boolean>;
+  error$: Observable<any>;
+  valid$: Observable<boolean>;
+  changed$: Observable<boolean>;
+  campaignName$: Observable<string>;
   routeSub: Subscription;
 
   constructor(
     private route: ActivatedRoute,
-    private router: Router,
-    private toastr: ToastrService,
-    public campaignStoreService: CampaignStoreService,
-    public store: Store<any>,
+    private store: Store<any>,
+    private campaignAction: CampaignActionService,
     private accountService: AccountService,
     private advertiserService: AdvertiserService
   ) {}
 
   ngOnInit() {
-    this.campaignSaving$ = this.store.pipe(select(selectCampaignSaving));
     this.routeSub = this.route.paramMap
```

Comment:
This is still a troublesome spot here that I do not love. From what I recall, we had issues with those select fields not populating with the loaded values if the options were not loaded before the form initialized?

Accounts and Advertisers could go into NgRx and be handled with campaign load on route change. Still some work to do to get all this cleaned up.